webpack/module-minifier-plugin/src/ModuleMinifierPlugin.ts

```diff
@@ -152,329 +170,352 @@ export class ModuleMinifierPlugin implements webpack.Plugin {
-                    const sourceForMap: string = `// ${mod.readableIdentifier(
-                      requestShortener
-                    )}${wrappedCode.slice(MODULE_WRAPPER_PREFIX.length, -MODULE_WRAPPER_SUFFIX.length)}`;
-
-                    const { code: minified, map: minifierMap, extractedComments } = result;
-
-                    const rawOutput: Source = useSourceMaps
-                      ? new SourceMapSource(
-                          minified, // Code
-                          nameForMap, // File
-                          minifierMap!, // Base source map
-                          sourceForMap, // Source from before transform
-                          map!, // Source Map from before transform
-                          false // Remove original source
-                        )
-                      : new RawSource(minified);
-
-                    const unwrapped: ReplaceSource = new ReplaceSource(rawOutput);
-                    const len: number = minified.length;
-
-                    unwrapped.replace(0, MODULE_WRAPPER_PREFIX.length - 1, '');
-                    unwrapped.replace(len - MODULE_WRAPPER_SUFFIX.length, len - 1, '');
-
-                    const withIds: Source = postProcessCode(unwrapped, mod.identifier());
-
-                    minifiedModules.set(realId, {
-                      source: new CachedSource(withIds),
-                      extractedComments,
-                      module: mod
-                    });
-                  } catch (err) {
-                    compilation.errors.push(err);
+        // During code generation, send the generated code to the minifier and replace with a placeholder
+        compilation.moduleTemplates.javascript.hooks.package.tap(TAP_AFTER, minifyModule);
+
+        // This should happen before any other tasks that operate during optimizeChunkAssets
+        compilation.hooks.optimizeChunkAssets.tapPromise(
+          TAP_BEFORE,
+          async (chunks: webpack.compilation.Chunk[]): Promise<void> => {
+            // Still need to minify the rendered assets
+            for (const chunk of chunks) {
+              const externals: string[] = [];
+              const externalNames: Map<string, string> = new Map();
+
+              const chunkModuleSet: Set<string | number> = new Set();
+              const allChunkModules: Iterable<IExtendedModule> =
+                chunk.modulesIterable as Iterable<IExtendedModule>;
+              let hasNonNumber: boolean = false;
+              for (const mod of allChunkModules) {
+                if (mod.id !== null) {
+                  if (typeof mod.id !== 'number') {
+                    hasNonNumber = true;
+                  }
+                  chunkModuleSet.add(mod.id);
+
+                  if (mod.external) {
+                    const key: string = `__WEBPACK_EXTERNAL_MODULE_${webpack.Template.toIdentifier(
+                      `${mod.id}`
+                    )}__`;
```

Comment:
Is this a const exported by webpack?

Reply:
Added a link to the webpack source code it is matching the behavior of.